spec/classes/web_spec.rb

```diff
@@ -53,52 +54,48 @@
-                             end
-                           when 'Debian'
-                             if facts[:operatingsystemmajrelease].to_i >= 9
-                               'php-pgsql'
-                             else
-                               'php5-pgsql'
-                             end
+        describe "with database_type as postgresql and zabbix_version #{zabbix_version}" do
+          let :params do
+            super().merge(database_type: 'postgresql')
+            super().merge(zabbix_version: zabbix_version)
+          end
+
+          pgsqlpackage = case facts[:operatingsystem]
+                         when 'Ubuntu'
+                           if facts[:operatingsystemmajrelease] >= '16.04'
+                             'php-pgsql'
+                           else
+                             'php5-pgsql'
+                           end
+                         when 'Debian'
+                           if facts[:operatingsystemmajrelease].to_i >= 9
+                             'php-pgsql'
                            else
                              'php5-pgsql'
                            end
-
-            packages = if facts[:osfamily] == 'RedHat'
-                         if facts[:operatingsystemmajrelease].to_i == 7 &&
-                            !%w[VirtuozzoLinux OracleLinux Scientific].include?(facts[:os]['name']) &&
-                            zabbix_version == '5.0'
-                           ['zabbix-web-pgsql-scl', 'zabbix-web']
                          else
-                           ['zabbix-web-pgsql', 'zabbix-web']
+                           'php5-pgsql'
                          end
+
+          packages = if facts[:osfamily] == 'RedHat'
+                       if facts[:operatingsystemmajrelease].to_i == 7 &&
+                          !%w[VirtuozzoLinux OracleLinux Scientific].include?(facts[:os]['name']) &&
+                          zabbix_version == '5.0'
+                         ['zabbix-web-pgsql-scl', 'zabbix-web']
                        else
-                         ['zabbix-frontend-php', pgsqlpackage]
+                         ['zabbix-web-pgsql', 'zabbix-web']
                        end
+                     else
+                       ['zabbix-frontend-php', pgsqlpackage]
+                     end
 
-            packages.each do |package|
-              it { is_expected.to contain_package(package) }
-            end
-            it { is_expected.to contain_file('/etc/zabbix/web/zabbix.conf.php').with_content(%r{^\$DB\['TYPE'\]     = 'POSTGRESQL'}) }
+          packages.each do |package|
+            it { is_expected.to contain_package(package) }
           end
+          it { is_expected.to contain_file('/etc/zabbix/web/zabbix.conf.php').with_content(%r{^\$DB\['TYPE'\]     = 'POSTGRESQL'}) }
         end
 
-        describe 'with database_type as mysql' do
+        describe 'with database_type as mysql', unless: facts[:os]['release']['major'] == '7' do
```

Comment:
I'm generally not a fan of checking a major release without the OS family/name,

Reply:
will update it

Reply:
done